The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6009027442/job/16297681574) [(plain)](https://github.com/rust-lang/rust/commit/e01facbcfd53ed16435818e55e06f4b30cb45f58/checks/16297681574/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=reez12g
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_50d12243-a325-4345-a241-574c6084112b
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=11af8873d93ddd7115b9b24b16514eb0f77317c3
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_50d12243-a325-4345-a241-574c6084112b
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_50d12243-a325-4345-a241-574c6084112b
GITHUB_TRIGGERING_ACTOR=reez12g
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/115335/merge
GITHUB_WORKFLOW_SHA=11af8873d93ddd7115b9b24b16514eb0f77317c3
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_18_X64=/opt/hostedtoolcache/go/1.18.10/x64
---

error[E0277]: can't compare `u64` with `usize`
   --> compiler/rustc_hir_typeck/src/generator_interior/mod.rs:647:46
    |
647 |             let plural_len = if target_usize == usize::MAX {
    |                                              ^^ no implementation for `u64 == usize`
    |
    = help: the trait `std::cmp::PartialEq<usize>` is not implemented for `u64`
    = help: the following other types implement trait `std::cmp::PartialEq<Rhs>`:
              <u64 as std::cmp::PartialEq>
              <u64 as std::cmp::PartialEq<rustc_target::json::Json>>
Some errors have detailed explanations: E0277, E0308, E0425.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `rustc_hir_typeck` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
```

</details>

--- a/compiler/rustc_hir_typeck/src/generator_interior/mod.rs
+++ b/compiler/rustc_hir_typeck/src/generator_interior/mod.rs
@@ -643,15 +643,15 @@ fn check_must_not_suspend_ty<'tcx>(
         }
         ty::Array(ty, len) => {
             let descr_pre = &format!("{}array{} of ", data.descr_pre, plural_suffix);
+            let target_usize = len.try_eval_target_usize(fcx.tcx, fcx.param_env).unwrap_or(0) as usize;
+            let plural_len = target_usize.saturating_add(1);
             check_must_not_suspend_ty(
                 fcx,
                 ty,
                 hir_id,
                 SuspendCheckData {
                     descr_pre,
-                    plural_len: len.try_eval_target_usize(fcx.tcx, fcx.param_env).unwrap_or(0)
-                        as usize
-                        + 1,
+                    plural_len,
                     ..data
                 },
             )

